src/discussions/comments/CommentsView.jsx

```diff
@@ -10,25 +10,27 @@ import { Button, Spinner } from '@edx/paragon';
 import { selectThread } from '../posts/data/selectors';
 import { markThreadAsRead } from '../posts/data/thunks';
 import Post from '../posts/post/Post';
-import { selectThreadComments } from './data/selectors';
-import { fetchThreadComments } from './data/thunks';
+import { courseSettingsSchemeDivided, selectThreadComments } from './data/selectors';
+import { fetchCourseSettings, fetchThreadComments } from './data/thunks';
 import Reply from './reply/Reply';
 import messages from './messages';
 
 ensureConfig(['POST_MARK_AS_READ_DELAY'], 'Comment thread view');
 
 function CommentsView({ intl }) {
-  const { postId } = useParams();
+  const { courseId, postId } = useParams();
   const dispatch = useDispatch();
   const thread = useSelector(selectThread(postId));
   const comments = useSelector(selectThreadComments(postId));
+  const courseSchemeDevided = useSelector(courseSettingsSchemeDivided);
   useEffect(() => {
     dispatch(fetchThreadComments(postId));
     const markReadTimer = setTimeout(() => {
       if (thread && !thread.read) {
         dispatch(markThreadAsRead(postId));
       }
     }, getConfig().POST_MARK_AS_READ_DELAY);
+    dispatch(fetchCourseSettings(courseId));
```

Comment:
We should fetch this in the discussions home component since it will be used by all, and only needs to be loaded once.